File: tests/fixtures/patterns-positive/proxy/mutB.py
# POSITIVO (mutación B: guarda de inicialización perezosa duplicada en 2 métodos).
class ShapeService:
    def __init__(self):
        self.real = None

    def area(self):
        if self.real is None:
            self.real = RealShape()
        return self.real.area()

    def paint(self, color):
        if self.real is None:
            self.real = RealShape()
        self.real.paint(color)

File: tests/fixtures/patterns-positive/proxy/mutB.ts
// POSITIVO (mutación B: guarda de inicialización perezosa duplicada en 2 métodos).
class ShapeService {
  private real: RealShape | null = null;
  area(): number {
    if (this.real === null) {
      this.real = new RealShape();
    }
    return this.real.area();
  }
  paint(color: string): void {
    if (this.real === null) {
      this.real = new RealShape();
    }
    this.real.paint(color);
  }
}
